grammar.js

```diff
@@ -319,3 +325,17 @@ function singleKeywordDirective($, keyword) {
             '\n',
     )
 }
+
+function block($, keyword) {
+    return seq(
+            token(keyword),
+            optional(seq($.whitespace, /.*/)),
+            '\n',
+            repeat(seq(optional(seq(optional($.whitespace), /.*/)), '\n')),
+            choice(
```

Comment:
I'm merging this as-is as ledger seems to accept all kinds of `end` directives, but I'm not a fan, because the documentation is stricter. Unless you have a use-case for naked or mismatched `end`s, I suggest that we remove it (I'm happy to do it myself). This way, we don't encourage users to use undocumented yet de-facto valid forms.

Concretly, that means nuking the `choice` and only keeping `token(\`end ${keyword}\`)`

Reply:
I had looked into creating a function that would handle `end` statements, but haven't gotten it to work yet.

How about creating an issue for this, so it is tracked?

Reply:
I don't think that we need another function, your `block` function is the right approach IMO.

Reply:
Sounds good and thanks for creating #27.